utils/bloom_filter.cc

```diff
@@ -121,5 +122,48 @@ filter_ptr create_filter(int hash, large_bitset&& bitset, filter_format format)
 filter_ptr create_filter(int hash, int64_t num_elements, int buckets_per, filter_format format) {
     return std::make_unique<murmur3_bloom_filter>(hash, large_bitset(get_bitset_size(num_elements, buckets_per)), format);
 }
+
+void maybe_fold_filter(filter_ptr& filter, int64_t num_elements, int buckets_per) {
+    auto bf = static_cast<bloom_filter*>(filter.get());
+    auto curr_num_bits = bf->bits().size();
+    size_t ideal_num_bits = get_bitset_size(num_elements, buckets_per);
+
+    if (curr_num_bits <= ideal_num_bits * 1.5) {
```

Comment:
If you fold 1.5 by a factor of 2, you get a size of 0.75 of ideal. What's the implications wrt false positive chance?